spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java

```diff
@@ -107,28 +108,57 @@ private boolean needRefresh(ConfigStore store) {
                 .withLabels(store.getLabels()).withFields(QueryField.ETAG).withRange(0, 0);
 
         List<KeyValueItem> keyValueItems = configOperations.getRevisions(store.getName(), options);
+        
         if (keyValueItems.isEmpty()) {
             return false;
         }
 
         String etag = keyValueItems.get(0).getEtag();
-        if (firstTimeMap.get(store.getName()) == null) {
-            storeEtagMap.put(store.getName(), etag);
-            firstTimeMap.put(store.getName(), false);
+        if (firstTimeMap.get(store.getName() + "configuration") == null) {
+            storeEtagMap.put(store.getName() + "configuration", etag);
+            firstTimeMap.put(store.getName() + "configuration", false);
             return false;
         }
 
-        if (!etag.equals(storeEtagMap.get(store.getName()))) {
+        if (!etag.equals(storeEtagMap.get(store.getName() + "configuration"))) {
             LOGGER.trace("Some keys in store [{}] matching [{}] is updated, will send refresh event.",
                     store.getName(), watchedKeyNames);
-            storeEtagMap.put(store.getName(), etag);
+            storeEtagMap.put(store.getName() + "configuration", etag);
             RefreshEventData eventData = new RefreshEventData(watchedKeyNames);
             publisher.publishEvent(new RefreshEvent(this, eventData, eventData.getMessage()));
             return true; // Break early once a change is found
         }
 
         return false;
     }
+    
+    private boolean needRefreshFeatureFlag(ConfigStore store) {
```

Comment:
As most of the code blocks in the `needRefresh` and `needRefreshFeatureFlag` are similar, how about merge these 2 methods?